static/js/login_or_signup.js

```diff
@@ -45,75 +45,111 @@ var checkKeypressIsSpace = function(event) {
 };
 
 $(function() {
-    var $signupForm = $('#signup-form');
-
-    var submitHasBeenClicked = false;
-
     var displayNameField = 'displayName';
     var emailAddressField = 'emailAddress';
     var passwordField = 'password';
 
-    var fields = [
-        displayNameField,
-        emailAddressField,
-        passwordField,
-    ];
-
-    getInputFields($signupForm, fields).forEach(
-        ($field) => {
-            var field = $field.attr('name');
-
-            // continuously update validation message after failed submission
-            $field.on(
-                'input',
-                (event) => {
-                    if (submitHasBeenClicked) {
-                        populateValidationMessage($field);
-                    }
-                });
-
-            if (field !== passwordField) {
-                // restrict initial space character
-                $field.keypress(
+    var signupFormMetadata = {
+        $form: $('#signup-form'),
+        fields: [
+            displayNameField,
+            emailAddressField,
+            passwordField,
+        ],
+        submitHasBeenClicked: false,
+    };
+
+    var loginFormMetadata = {
+        $form: $('#login-form'),
+        fields: [
+            emailAddressField,
+            passwordField,
+        ],
+        submitHasBeenClicked: false,
+    };
+
+
+    var installFieldValidators = function(formMetadata) {
+        getInputFields(formMetadata.$form, formMetadata.fields).forEach(
+            ($field) => {
+                var field = $field.attr('name');
+
+                // continuously update validation message after failed submission
+                $field.on(
+                    'input',
                     (event) => {
```

Comment:
i think this was my bad, but i try to avoid having unused parameters. So i would delete `event`